### DISCUSSION TITLE
【firestore】Please help me. I can't get where and orderBy to work at the same time.

### DISCUSSION THREAD

**Discussion by namachan0219** [invertase/react-native-firebase#5417] [Q&A]
Hello everyone.
I have one question for you.

When I combine where and orderBy in firestore onSnapshot, the data subscription is not executed.

○ onSnapshot & orderBy => subscribe working
✗ onSnapshot & where => subscribe not working
✗ onSnapshot & where & orderBy  => subscribe not working

Can someone please tell me how to solve this?

[Data in the firestore collection]
※ createdAt / updateAt → Originally, it has a date.
    {
      name: "data1",
      type: "trends",
      updateAt: firestore.FieldValue.serverTimestamp(),
      createdAt: firestore.FieldValue.serverTimestamp(),
    },
    {
      name: "data2",
      type: "hashtag",
      updateAt: firestore.FieldValue.serverTimestamp(),
      createdAt: firestore.FieldValue.serverTimestamp(),
    },

[My code]
  useEffect(() => {
    const subscriber = firestore()
      .collection("Threads")
      .where("type", "==", "trends") or .where("type", "==", "hashtag") 
      .orderBy("updateAt", "desc")
      .limit(10)
      .onSnapshot((querySnapshot) => {
        ... Function to save to state....
      });
    return () => subscriber();
  }, [trigger]);

[Package.json]
{
    "react": "16.13.1",
    "react-dom": "16.13.1",
    "react-native": "~0.63.4",
    "@react-native-firebase/admob": "^11.5.0",
    "@react-native-firebase/analytics": "^11.5.0",
    "@react-native-firebase/app": "^11.5.0",
    "@react-native-firebase/auth": "^11.5.0",
    "@react-native-firebase/crashlytics": "^11.5.0",
    "@react-native-firebase/database": "^11.5.0",
    "@react-native-firebase/firestore": "^11.5.0",
    "@react-native-firebase/functions": "^11.5.0",
    "@react-native-firebase/perf": "^11.5.0",
    "@react-native-firebase/storage": "^11.5.0",
}

Love from Japan to the community.

**Comment by andersonaddo** [comment]
Not sure about why using a single "where" isn't working, but for the third option (when you combine everything), you're gonna need a compound index
https://firebase.google.com/docs/firestore/query-data/index-overview

**Reply by namachan0219** [reply]
@andersonaddo 

Thank you for your reply.

In firebase, the composite index of type and updateAt is
"desc" is enabled.

This is exactly the same situation as Issues #1573.
I don't know if it is a bug in rnfb or my environment is bad.
I'm very worried about this.

I hope you can tell me more.

**Reply by mikehardy** [reply]
@namachan0219 please open a new issue with the template details filled in, using up to date versions (12.1.0 of this package, and that should pull in the most up to date underlying SDKs, they should not be overridden so we know we are current...) and an App.js that is a minimal viable reproduction https://stackoverflow.com/help/minimal-reproducible-example when dropped in to the result of a clean demo app from https://github.com/mikehardy/rnfbdemo/blob/master/make-demo.sh - otherwise we're just guessing.

You may like to examine our E2E tests to see if we exercise your case https://github.com/invertase/react-native-firebase/blob/master/packages/firestore/e2e/ - if we do not then an alternate path to a minimal viable reproduction is to fork/clone the repo, bring up the tests app (https://github.com/invertase/react-native-firebase/blob/master/tests/README.md) and add a test case that should work but fails, then point us to it.

**Reply by namachan0219** [reply]
@mikehardy 

Hello.Thank you for your reply.

As of now, I have used get instead of onSnapshot as an alternative.

I'll check the information you provided when I get settled.
I will report back to the community when I can confirm the issue.
